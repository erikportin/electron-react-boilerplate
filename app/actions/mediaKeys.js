// @flow
export const PLAY_PAUSE_ACTIVATED = 'PLAY_PAUSE_ACTIVATED';
export const PLAY_PAUSE_DEACTIVATED = 'PLAY_PAUSE_DEACTIVATED';

type actionType = {
  +type: string
};

export function activate() {
  return {
    type: PLAY_PAUSE_ACTIVATED
  };
}

export function deactivate() {
  return {
    type: PLAY_PAUSE_DEACTIVATED
  };
}
