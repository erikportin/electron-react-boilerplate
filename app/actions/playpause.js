// @flow
import type { counterStateType } from '../reducers/counter';

type actionType = {
  +type: string
};

export const PLAY_PAUSE = 'PLAY_PAUSE';

export function playPause() {
  return {
    type: PLAY_PAUSE
  };
}