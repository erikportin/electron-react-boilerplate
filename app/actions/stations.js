// @flow
export const ADD_STATION = 'ADD_STATION';
export const REMOVE_STATION = 'REMOVE_STATION';

type actionType = {
  +type: string
};

export function add() {
  return {
    type: ADD_STATION
  };
}

export function remove() {
  return {
    type: REMOVE_STATION
  };
}
