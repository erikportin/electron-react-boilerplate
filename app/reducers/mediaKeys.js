// @flow
import { PLAY_PAUSE_DEACTIVATED, PLAY_PAUSE_ACTIVATED } from '../actions/mediaKeys';

export type playPauseKeyActiveStateType = {
  +playPauseKeyActive: boolean
};

type actionType = {
  +type: string
};

export default function playPauseKeyActive(state: boolean = false, action: actionType) {
  switch (action.type) {
    case PLAY_PAUSE_DEACTIVATED:
      return false;
    case PLAY_PAUSE_ACTIVATED:
      return true;
    default:
      return state;
  }
}
