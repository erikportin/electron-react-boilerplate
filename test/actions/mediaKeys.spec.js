import * as actions from '../../app/actions/mediaKeys';

describe('actions', () => {
  it('should activate should create activate action', () => {
    expect(actions.activate()).toMatchSnapshot();
  });

  it('should deactivate should create deactivate action', () => {
    expect(actions.deactivate()).toMatchSnapshot();
  });
});
