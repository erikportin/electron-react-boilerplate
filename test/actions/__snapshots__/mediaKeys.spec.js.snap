// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actions should activate should create activate action 1`] = `
Object {
  "type": "PLAY_PAUSE_ACTIVATED",
}
`;

exports[`actions should deactivate should create deactivate action 1`] = `
Object {
  "type": "PLAY_PAUSE_DEACTIVATED",
}
`;
